subsys/bluetooth/mesh/Kconfig

```diff
@@ -231,6 +231,13 @@ config BT_MESH_PB_ADV_RETRANS_TIMEOUT
 	help
 	  Timeout value of retransmit provisioning PDUs.
 
+config BT_MESH_PB_ADV_DELAYABLE_ADV_COUNT
```

Comment:
Do we need this configurable? How do people should figure out how many advs they need for the provisioning to work correctly? Will it make any difference if they use 5 instead of 4 buffers? Won't they brake the standard by decreasing this value (random delay is a shall statement in the spec)?

Reply:
I don't. But I don't understand how users will figure out what value they need to set to be sure that the provisioning always works and doesn't violate the spec.

From what I see, the Transaction Start, Transaction Continuation and Link Open uses `link.tx.adv[]` to keep allocated adv buffers. All these messages don't use `bt_mesh_send_cb` callback as they use `send_reliable` function.

For the Link Ack, Link Close and Transaction Ack allocated adv buffer isn't stored anywhere. They also use `bt_mesh_send_cb` callback (Link Ack actually may not need the callback). All these 3 messages can't be sent at the same time: Link Ack is sent only when link is not opened yet (thus no Transaction PDUs can't be sent yet), Link Close is sent at the end of the provisioning (thus no Transaction PDUs can't be sent after that). This means we just need one extra pointer to adv buffer (+ `bt_mesh_send_cb` and `cb_data`). I see one corner case here: if we are sending Transaction Ack (or Link Ack) when triggering Link Close, but I don't see much problem with this.

If didn't miss anything, we just need extra adv buffer (+ `bt_mesh_send_cb` and `cb_data`) for Link Ack, Link Close and Transacation Ack messages and a `k_work_delayable` that checks if this extra adv buffer or `link.tx.adv[]` are set and sends them one by one with random delay.

One option to avoid storing `bt_mesh_send_cb` callback and `cb_data` is to change the advertiser api to allow to set the callback and the data when allocating an adv buffer.

Reply:
I look at that I did not think it was a better idea. We need a way if keeping track of which link.tx.adv that has been sent so we need a seperate list than link.tx.adv, because we can not delete the link.tx.adv when it is sent because we need to keep it because of retransmission. So then we need a seperate mechanise of keeping track of that is going to be sent. For the ack and the sending the unack we need also to keep the pointers to the adv buffer and the callback and cb_data. So that has also to be made into seprate entries. 
I think it is best to keep this into one array/list for all of them. That makes it the easiest what to maintain and understand what it is use for. It can also be recognized from other parts of the code.

Reply:
I do not see that what you suggest is a better solution.